test/inspector_controller_test.dart

```diff
@@ -0,0 +1,820 @@
+            '    text: "Hello, World!"\n'
+            '    ▼renderObject: RenderParagraph#00000 relayoutBoundary=up2\n'
+            '      parentData: offset=Offset(360.5, 264.0) (can use size)\n'
+            '      constraints: BoxConstraints(0.0<=w<=800.0, 0.0<=h<=544.0)\n'
+            '      size: Size(79.0, 16.0)\n'
+            '      textAlign: start\n'
+            '      textDirection: ltr\n'
+            '      softWrap: wrapping at box width\n'
+            '      overflow: clip\n'
+            '      locale: en_US\n'
+            '      maxLines: unlimited\n'),
+      );
+
+      // Select row nine (right text)
+      detailsTree.onTap(const Offset(0, rowHeight * 9.5));
+      expect(
+        detailsTree.toStringDeep(),
+        equalsIgnoringHashCodes(
+          '▼[/icons/inspector/textArea.png] Text \n'
+              '│   "Hello, World!"\n'
+              '│   textAlign: null [D]\n'
+              '│   textDirection: null [D]\n'
+              '│   locale: null [D]\n'
+              '│   softWrap: null [D]\n'
+              '│   overflow: null [D]\n'
+              '│   textScaleFactor: null [D]\n'
+              '│   maxLines: null [D]\n'
+              '└─▼[/icons/inspector/textArea.png] RichText  <-- selected\n'
+              '    softWrap: wrapping at box width\n'
+              '    maxLines: unlimited\n'
+              '    text: "Hello, World!"\n'
+              '    ▼renderObject: RenderParagraph#00000 relayoutBoundary=up2\n'
+              '      parentData: offset=Offset(360.5, 264.0) (can use size)\n'
+              '      constraints: BoxConstraints(0.0<=w<=800.0, 0.0<=h<=544.0)\n'
+              '      size: Size(79.0, 16.0)\n'
+              '      textAlign: start\n'
+              '      textDirection: ltr\n'
+              '      softWrap: wrapping at box width\n'
+              '      overflow: clip\n'
+              '      locale: en_US\n'
+              '      maxLines: unlimited\n',
+        ),
+      );
+
+      // make sure the main tree didn't change.
+      expect(tree.toStringDeep(), equalsIgnoringHashCodes(textSelected));
+
+      // select row index 3.
+      tree.onTap(const Offset(0, rowHeight * 3.5));
+      expect(
+          tree.toStringDeep(),
+          equalsIgnoringHashCodes(
+            '▼[[] root ]\n'
+                '└─▼[M] MyApp \n'
+                '  └─▼[M] MaterialApp \n'
+                '    └─▼[S] Scaffold  <-- selected\n'
+                '      ├───▼[C] Center \n'
+                '      │   └─▼[/icons/inspector/textArea.png] Text \n'
+                '      └─▼[A] AppBar \n'
+                '        └─▼[/icons/inspector/textArea.png] Text \n',
+          ));
+
+      await detailsTree.nextUiFrame;
+      // This tree is huge. If there is a change to package:flutter it may
+      // change. If this happens don't panic and rebaseline the content.
+      expect(
+        detailsTree.toStringDeep(),
+        equalsIgnoringHashCodes(
+          '▼[S] Scaffold  <-- selected\n'
```

Comment:
Should we keep this golden output in a txt file?

Reply:
I'm torn about this. On one hand a text file makes rebaselining easy but on the other hand inline makes the intent of the test much clearer. As a middle ground I plan to fixup the output here so most of these tests are truncated output only highlighting the relevant details so rebaselining won't be needed much or at all and the forest is easier to see from the trees.
note that there are a number of inline golden style text output tests in flutter proper and they rarely break.